feature/experimental/gribi/otg_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,703 @@
+	t.Helper()
+	var primary, secondary string
+	for _, controller := range controllers {
+		role := gnmi.Get(t, dut, gnmi.OC().Component(controller).RedundantRole().State())
+		t.Logf("Component(controller).RedundantRole().Get(t): %v, Role: %v", controller, role)
+		switch role {
+		case secondaryController:
+			secondary = controller
+		case primaryController:
+			primary = controller
+		default:
+			t.Fatalf("Expected controller %s to be active or standby, got %v", controller, role)
+		}
+	}
+	if secondary == "" || primary == "" {
+		t.Fatalf("Expected non-empty primary and secondary Controller, got primary: %v, secondary: %v", primary, secondary)
+	}
+	t.Logf("Detected primary: %v, secondary: %v", primary, secondary)
+
+	return secondary, primary
+}
+
+// switchoverReady is to check if controller is ready for switchover.
+func switchoverReady(t *testing.T, dut *ondatra.DUTDevice, controller string) {
+	switchoverReady := gnmi.OC().Component(controller).SwitchoverReady()
+	gnmi.Await(t, dut, switchoverReady.State(), 30*time.Minute, true)
+}
+
+// validateSwitchoverTelemetry validates telemetry sensors.
+func validateSwitchoverTelemetry(t *testing.T, dut *ondatra.DUTDevice, primaryAfterSwitch string) {
+	t.Helper()
+	t.Log("Validate OC Switchover time/reason.")
+	primary := gnmi.OC().Component(primaryAfterSwitch)
+	if !gnmi.Lookup(t, dut, primary.LastSwitchoverTime().State()).IsPresent() {
+		t.Errorf("primary.LastSwitchoverTime().Lookup(t).IsPresent(): got false, want true.")
+	} else {
+		t.Logf("Found primary.LastSwitchoverTime(): %v", gnmi.Get(t, dut, primary.LastSwitchoverTime().State()))
+	}
+
+	if !gnmi.Lookup(t, dut, primary.LastSwitchoverReason().State()).IsPresent() {
+		t.Errorf("primary.LastSwitchoverReason().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastSwitchoverReason := gnmi.Get(t, dut, primary.LastSwitchoverReason().State())
+		t.Logf("Found lastSwitchoverReason.GetDetails(): %v", lastSwitchoverReason.GetDetails())
+		t.Logf("Found lastSwitchoverReason.GetTrigger().String(): %v", lastSwitchoverReason.GetTrigger().String())
+	}
+}
+
+// checkNIHasNEntries uses the Get RPC to validate that the network instance named ni
+// contains want (an integer) entries.
+func checkNIHasNEntries(ctx context.Context, c *fluent.GRIBIClient, ni string, t testing.TB) int {
+	t.Helper()
+	gr, err := c.Get().
+		WithNetworkInstance(ni).
+		WithAFT(fluent.AllAFTs).
+		Send()
+
+	if err != nil {
+		t.Fatalf("Unexpected error from get, got: %v", err)
+	}
+	return len(gr.GetEntry())
+}
+
+// Send traffic and validate traffic.
+func testTraffic(t *testing.T, args testArgs) {
+	t.Helper()
+	args.ate.OTG().StartTraffic(t)
+	// Send traffic for 1 minute.
+	time.Sleep(1 * time.Minute)
+	args.ate.OTG().StopTraffic(t)
+	otgutils.LogFlowMetrics(t, args.ate.OTG(), args.top)
+
+	txPkts := float32(gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow("Flow").Counters().OutPkts().State()))
```

Comment:
Change to a single query like in https://github.com/openconfig/featureprofiles/blob/e8068e3f70aeb6cbcbd454bd72337106e76c473c/feature/gribi/otg_tests/supervisor_failure_test/supervisor_failure_test.go#L163

Reply:
Done